ntbs-service/Jobs/UpdateTableCountsJob.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Hangfire.Console;
+using Hangfire.Server;
+using Newtonsoft.Json;
+using ntbs_service.DataAccess;
+using Serilog;
+
+namespace ntbs_service.Jobs
+{
+    public class UpdateTableCountsJob
+    {
+        private readonly ITableCountsRepository _tableCountsRepository;
+        private PerformContext _context;
+
+        public UpdateTableCountsJob(ITableCountsRepository tableCountsRepository)
+        {
+            _tableCountsRepository = tableCountsRepository;
+        }
+
+        /// PerformContext context is passed in via Hangfire Server
+        public async Task Run(PerformContext context)
+        {
+            _context = context;
+            Log.Information("Starting table counting job.");
+
+            var results = await _tableCountsRepository.ExecuteUpdateTableCountsStoredProcedure();
+            CheckExecutedSuccessfully(results);
+
+            Log.Information("Starting table count comparison.");
+
+            await CompareRecentTableCounts();
+
+            Log.Information("Finishing table counting job.");
+        }
+
+        private async Task CompareRecentTableCounts()
+        {
+            var counts = (await _tableCountsRepository.GetRecentTableCounts()).ToList();
+
+            if (counts.Count < 2)
+            {
+                throw new ApplicationException(
+                    "Not enough table counts available. Try re-running the update-table-counts job.");
+            }
+
+            var currentCount = counts[0];
+            var previousCount = counts[1];
+
+            var messages = currentCount
+                .GetType()
+                .GetProperties()
+                .Where(property => property.PropertyType == typeof(int))
+                .Select(property => new
+                {
+                    property,
+                    currentValue = (int)property.GetValue(currentCount),
+                    previousValue = (int)property.GetValue(previousCount)
+                })
+                .Where(propertyValues => propertyValues.currentValue < propertyValues.previousValue)
+                .Select(propertyValues =>
+                    $"Property: {propertyValues.property.Name} has decreased " +
+                    $"from {propertyValues.previousValue} to {propertyValues.currentValue}")
+                .ToList();
+
+            if (messages.Any())
+            {
+                var message = $"Some table counts have decreased:\n{string.Join("\n", messages)}";
+
+                Log.Warning(message);
+                _context.WriteLine(message);
```

Comment:
Yeah I was unsure how to report this back nicely, I chose warning because none of the code had outright failed, but an exception will be more eye-catching. I'll use that instead